hub/core/chunk_engine/write.py

```diff
@@ -14,17 +14,32 @@
 from .read import read_index_map, read_tensor_meta, tensor_exists
 
 
+def tuple_to_list(shape: Tuple):
+    shapeArray = np.asarray(shape)
+    return shapeArray.tolist()
+
+
 def write_tensor_meta(key: str, storage: StorageProvider, meta: dict):
-    storage[get_tensor_meta_key(key)] = pickle.dumps(meta)
+    meta["min_shape"] = tuple_to_list(meta["min_shape"])
+    meta["max_shape"] = tuple_to_list(meta["max_shape"])
+    storage[get_tensor_meta_key(key)] = bytes(json.dumps(meta), "utf-8")
 
 
 def write_index_map(key: str, storage: StorageProvider, index_map: list):
     index_map_key = get_index_map_key(key)
-    storage[index_map_key] = pickle.dumps(index_map)
+    storage[index_map_key] = bytes(json.dumps(index_map), "utf-8")
 
 
 def write_dataset_meta(storage: StorageProvider, meta: dict):
-    storage[META_FILENAME] = pickle.dumps(meta)
+    storage[META_FILENAME] = bytes(json.dumps(meta), "utf-8")
+
+
+def dict_to_list(dictionary: dict):
+    dictlist = []
+    for key, value in dictionary.items():
+        temp = [key, value]
+        dictlist.append(temp)
+    return dictlist
```

Comment:
`dict.items` is a method that returns an iterable over the pairs of elements.

Reply:
Thank you for the suggestion @benchislett, will make this change!